### DISCUSSION TITLE
web worker with cdn occurs cors error

### DISCUSSION THREAD

**Discussion by Hyodori04** [webpack/webpack#17951] [Q&A]
I'm using web worker like below code
`        new Worker(new URL("./hash.worker.ts", import.meta.url)),
`

It works well in local. But when apply to production, it makes cors error

` Uncaught (in promise) DOMException: Failed to construct 'Worker': Script at  '{cdn-url}'
cannot be accessed from origin '{production-url}'`

I have read some discussion about this problem and applied solution that they recomment
But it dosen't work. 

Is there any way to solve this problem?

**Comment by alexander-akait** [comment]
Please provide example of your structure, it is not a problem of webpack, anyway if you provide reproducible example (for example), I will provide good solutions

**Reply by Hyodori04** [reply]
```
hash.worker.ts

import { md5 } from "hash-wasm";
import { HASH_ERROR } from "./useInputImageHandler/constants";

export const onMessage = async (e: MessageEvent<any>) => {
  try {
    const hash = await md5(e.data);
    self.postMessage(hash);
  } catch {
    self.postMessage(HASH_ERROR);
  }
};

self.onmessage = onMessage;
```

```
new Worker(new URL("./hash.worker.ts", import.meta.url))
```

```
const nextConfig = {
  ...
  assetPrefix: "https://cdn.front"
  ...
```


Worker file works well in local. Same origin with service and script source
But it become problem if used in production because service domain and script source is diffrent (because script is served by cdn)

**Reply by alexander-akait** [reply]
Please try https://github.com/vercel/next.js/discussions/29415 or change your server options
